modules/core/src/test/java/org/apache/synapse/message/store/RabbitMQStoreTest.java

```diff
@@ -0,0 +1,156 @@
+        connectionFactory.setAccessible(true);
+        //Setting parameters of the RabbitMQStore
+        Map<String, Object> temp = new HashMap<>();
+        temp.put(RabbitMQStore.USERNAME, USERNAME);
+        temp.put(RabbitMQStore.PASSWORD, PASSWORD);
+        temp.put(RabbitMQStore.HOST_NAME, HOST);
+        temp.put(RabbitMQStore.HOST_PORT, PORT);
+        temp.put(RabbitMQStore.VIRTUAL_HOST, VIRTUAL_HOST);
+        temp.put(RabbitMQStore.QUEUE_NAME, QUEUE);
+        rabbitMQStore.setParameters(temp);
+        AxisConfiguration axisConfiguration = new AxisConfiguration();
+        configurationContext = new ConfigurationContext(axisConfiguration);
+        SynapseConfiguration synapseConfiguration = new SynapseConfiguration();
+        Axis2SynapseEnvironment env = new Axis2SynapseEnvironment(configurationContext, synapseConfiguration);
+        rabbitMQStore.init(env);
+    }
+
+    /**
+     * call init method with dummy values and validating connectionFactory object
+     *
+     * @throws NoSuchFieldException
+     * @throws IllegalAccessException
+     */
+    @Test
+    public void testInit() throws NoSuchFieldException, IllegalAccessException {
+        ConnectionFactory factory = (ConnectionFactory) connectionFactory.get(rabbitMQStore);
+        Assert.assertEquals("should return previously stored values", factory.getPort(), Integer.parseInt(PORT));
+        Assert.assertEquals("should return previously stored values", factory.getHost(), HOST);
+        Assert.assertEquals("should return previously stored values", factory.getPassword(), PASSWORD);
+        Assert.assertEquals("should return previously stored values", factory.getUsername(), USERNAME);
+        Assert.assertEquals("should return previously stored values", factory.getVirtualHost(), VIRTUAL_HOST);
+    }
+
+    /**
+     * Creating a producer and validate initialized
+     */
+    @Test
+    public void testGetProducer() {
+        RabbitMQProducer messageProducer = (RabbitMQProducer) rabbitMQStore.getProducer();
+        Assert.assertTrue("MessageProducer should be initialized", messageProducer.isInitialized());
+    }
+
+    /**
+     * Creating a consumer and validating consumer id
+     */
+    @Test
+    public void testGetConsumer() {
+        RabbitMQConsumer messageConsumer = (RabbitMQConsumer) rabbitMQStore.getConsumer();
+        Assert.assertEquals("Comparing MessageConsumer id", "[null-C-1]", messageConsumer.getId());
+    }
+
+    /**
+     * Create axis2MessageContext
+     */
+    @Test
+    public void testNewAxis2Mc() {
+        MessageContext messageContext = rabbitMQStore.newAxis2Mc();
+        Assert.assertEquals("MessageContext should be initiated with configurationContext",messageContext
+                .getConfigurationContext(), configurationContext);
+    }
+
+    //Following test cases assert currently not implemented methods which are just returning null
+    @Test
```

Comment:
Describe the following test cases

Reply:
Methods tested in these tests are not implemented yet. I add these test case for coverage purposes and for the ease of changing them when methods are actually implemented.